design/crd-status.md

```diff
@@ -0,0 +1,312 @@
+        bgpStatus string
+        bfdStatus string
+    }
+
+    type BGPSessionStatus struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+        Status            MetalLBBGPStatus `json:"status,omitempty"`
+    }
+```
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: BGPSessionStatus
+metadata:
+  name: worker0-peer1
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/peer: peer1
+status:
+    bgpStatus: Established
+    bfdStatus: Up
+```
+
+The string exposed is taken directly from the output of FRR.
+
+If BFD is not configured for a given `BGPPeer`, the exposed bfdStatus will be "N/A".
+
+**A note about the implementation:** without entering to much into details,
+we will need to implement some sort of polling of the FRR status. Given the
+fact that this CR has no relation with the existing ones, a valid approach is
+to follow what was done for the [metrics exporter](https://github.com/metallb/metallb/blob/main/frr-metrics/exporter.go)
+and have a different component (or even the exporter itself) polling FRR and filling
+the session status. The polling interval must be configurable and large enough to
+avoid impacts both on FRR and on the API.
+The speaker should continue not to have direct interactions with FRR.
+
+#### Service Announcement - BGP
+
+Given a service and a node, we want to expose the `BGPpeer`s the service is configured
+to be advertised to.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: ServiceBGPStatus
+metadata:
+  name: service1-worker0
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/service: service1
+status:
+    bgpPeers:
+        - peerA
+        - peerB
+```
+
+```go
+    type MetalLBServiceBGPStatus struct {
+        Peers []string
+    }
+
+    type ServiceBGPStatus struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+        Status            MetalLBServiceBGPStatus `json:"status,omitempty"`
+    }
+```
+
+The labels will allow to easily discover which services are advertised from
+a given node, or all the peers a given service is advertised to.
+
+**Note:** this status won't take into consideration the status of the
+session with the given BGP peer, but only the will to advertise to that peer.
+This, to overcome the considerations about scalability written in the preface.
+
+#### Service Announcement - L2
+
+The useful information related to L2 are related to the node that is exposing
+the service, and via which interfaces.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2AdvertisementStatus
+metadata:
+  name: service1
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/service: service1
+status:
+    node: worker0
+    interfaces:
+        - eth0
+        - eth1
+```
+
+```go
+    type MetalLBServiceL2Status struct {
+        Node       string
+        Interfaces []string
```

Comment:
this might be useful, but I am afraid the information has to traverse too many layers. 
What I will do is, change the field type from []string to []InterfaceInfo or something like that so it's gonna be easier to extend